sympy/functions/combinatorial/numbers.py

```diff
@@ -2292,3 +2293,152 @@ def ok(x):
+#                                                                            #
+#                           Padovan numbers                                  #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class padovan(Function):
+    r"""
+    In number theory, the Padovan sequence is the sequence of integers P(n)
+    defined by the initial values `P_0 = 1`, `P_1 = 1` , `P_2 = 1` and the
+    the two-term recurrence `P_n = P_{n-2} + P_{n-3}`.
+
+    * ``padovan(n)`` gives the `n^{th}` Padovan number, `P_n`
+
+    Examples
+    ========
+
+    >>> from sympy import padovan
+
+    >>> padovan.is_padovan(2)
+    True
+    >>> [padovan(x) for x in range(0,11)]
+    [1, 1, 1, 2, 2, 3, 4, 5, 7, 9, 12]
+    >>> padovan.find_first_n_padovan_numbers(8)
+    [1, 1, 1, 2, 2, 3, 4, 5]
+    >>> padovan.find_padovan_numbers_in_range(2,39)
+    [2, 2, 3, 4, 5, 7, 9, 12, 16, 21, 28, 37]
+
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Padovan_sequence
+    .. [2] https://mathworld.wolfram.com/PadovanSequence.html
+
+    """
+
+    @staticmethod
+    def is_padovan(n):
+        n = sympify(n)
+        if n.is_Integer:
+            if n > 0:
+                if n == 1 :
+                    return True
+
+                sn0, sn1, sn2 = 1, 1, 1
+                while sn2 < n :
+                    b = sn0 + sn1
+                    sn0 = sn1
+                    sn1 = sn2
+                    sn2 = b
+
+                if sn2 == n:
+                    return True
+                else:
+                    return False
+
+            if n <= 0:
+                if n in (0, -1):
+                    return True
+
+                sn0, sn1, sn2 = 1, 0, 1
+                while sn2 > n :
+                    b = sn0 - sn2
+                    sn0 = sn1
+                    sn1 = sn2
+                    sn2 = b
+
+                if Eq(sn2, n, evaluate=True):
+                    return True
+                else:
+                    return False
+
+        else:
+            return False
+
+    @staticmethod
+    def find_first_n_padovan_numbers(n):
+        # Returns first n non-negative numbers from the padovan sequence
+        try:
+            n = as_int(n)
+        except ValueError:
+            raise ValueError('The provided number must be a positive integer')
+        if n < 0:
+            raise ValueError('The provided number must be a positive integer')
+        padovan = [1]
+        if n > 1:
+            padovan.append(1)
+
+        sn0, sn1, sn2 = 1, 1, 1
+        i = 2
+        while i < n :
+            padovan.append(sn2)
+            b = sn0 + sn1
+            sn0 = sn1
+            sn1 = sn2
+            sn2 = b
+            i = i + 1
+
+        return padovan
+
+    @staticmethod
+    def find_padovan_numbers_in_range(x, y):
```

Comment:
perhaps following `primerange` example, call it `padovanrange` and use same semantics of not including the end point?

Reply:
(At least leave off 'find_', IMO.)

Reply:
Thanks @smichr , I had actually followed naming schemes from the motzkin number sequence which was the latest number added to combinatorial .But your suggestion makes sense . I have made the required changes !

Reply:
@oscarbenjamin 
I had made some commits a while ago but tests haven't started running yet, which is strange .Is this some unknown mistake on my part ?